doc/dev/Mgmt-Naming-Conventions.md

```diff
@@ -0,0 +1,99 @@
+# Azure SDK for .NET - Management Plane SDK Naming Conventions
+
+This document describes the naming conventions for the Azure Management Plane SDK for .NET.
+In addition to adhering to the [General Naming Conventions of .NET](https://learn.microsoft.com/dotnet/standard/design-guidelines/general-naming-conventions), there are specific guidelines related to the naming of classes, properties, and methods. These include rules for using abbreviations and acronyms, as well as recommendations for avoiding Azure-specific names.
+
+## General Guidelines
+
+**DO** Use Pascal case which capital case for the first letter only.
+
+**DO** 2 letter acronyms are special which uppercase all if not part of longer compound identifiers, but still have some except like `Vm`, `Id` and etc.  
+
+**DO** Use a singular type name for an enumeration unless its values are bit fields. Use a plural type name for an enumeration with bit fields as values, also called flags enum. For more details, please check [here](https://docs.microsoft.com/dotnet/standard/design-guidelines/names-of-classes-structs-and-interfaces#naming-enumerations).
```

Comment:
```suggestion
**DO** Use a singular type name for an `enum` type if it does not have the `[Flags]` attribute. Use a plural type name for an `enum` type if it has `[Flags]` attribute. For more details, please check [here](https://docs.microsoft.com/dotnet/standard/design-guidelines/names-of-classes-structs-and-interfaces#naming-enumerations).
```

Reply:
I think putting the flag attribute here could be more explicit for the reader, despite we are copying things from the link.